make narrow combo popup readable

dterrahe opened the pull request:
As proposed here https://github.com/darktable-org/darktable/pull/11298#issuecomment-1095314021
make popup of a too narrow bauhaus combo wide enough to show label and entries.
Also force it to be on-screen even with narrow right panels or when called via shortcut while mouse is near screen edge.
![image](https://user-images.githubusercontent.com/1549490/162801425-443aff1c-b3a8-428b-a000-42f393fe6bc4.png)

![image](https://user-images.githubusercontent.com/1549490/162801374-9c04c813-8c29-44ef-bb46-fe8c4135f06d.png)

TurboGit commented:
We may want to be a bit more conservative. I have enough space for the combo, see:

![image](https://user-images.githubusercontent.com/467069/162906017-9da18514-6dd4-4490-9639-624925ffbce9.png)

But with this PR:

![image](https://user-images.githubusercontent.com/467069/162906386-00cb05d4-03c3-4e48-b225-a15c33656b31.png)

dterrahe commented:
Thanks for catching that!

The natural (minimal) width of the popup is different from the widget itself because the label is only shown on the first line (or not at all, if there are sections). Fixed this and also fixed the ellipsized "<" in the stars filter.

Removed `dt_bauhaus_combobox_set_popup_scale` (in collect.c), because it is not needed after this PR, and removed one of the three (3!) `scale` members of bauhaus.

TurboGit reviewed (approved):
Just perfect now, thanks!
